### PR TITLE
[BUG](view) fix can't create view with lambda function

### PR DESCRIPTION
**Pull request by zhangstar333**
## Proposed changes
before the lambda function Expr not implement toSqlImpl() function.
so it's call parent function, which is not suit for lambda function.
and will be have error when create view.

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by zhangstar333** [comment]
run buildall

**Comment by CalvinKirs** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/analysis/LambdaFunctionCallExpr.java [hunk 1]
```diff
@@ -263,4 +263,30 @@ protected void toThrift(TExprNode msg) {
...
     }
+
+    @Override
+    public String toSqlImpl() {
+        StringBuilder sb = new StringBuilder();
+        sb.append(getFnName().getFunction());
+        sb.append("(");
+        int childSize = children.size();
+        Expr lastExpr = getChild(childSize - 1);
+        boolean lastIsLambdaExpr = (lastExpr instanceof LambdaFunctionExpr);
```
Comment (HappenLee) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/LambdaFunctionCallExpr.java:274]: add comment for the code, trickly here

File: fe/fe-core/src/main/java/org/apache/doris/analysis/FunctionCallExpr.java [hunk 2]
```diff
@@ -766,7 +766,14 @@ protected void toThrift(TExprNode msg) {
             }
             msg.setAggExpr(aggFnParams.createTAggregateExpr(isMergeAggFn));
         } else {
-            msg.node_type = TExprNodeType.FUNCTION_CALL;
+            // array_filter(array, boolean) only implement as lambda function
+            // eg: SELECT array_filter(`array_d`, array_map(x -> (x >= 1), `array_d`))
+            // can't as a normal function for BE, need handle by lambda function call expr
+            if (fnName.getFunction().equalsIgnoreCase("array_filter")) {
```
Comment (HappenLee) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/FunctionCallExpr.java:772]: better BE support function "array_filter" as function call

**Comment by zhangstar333** [comment]
run buildall

**Comment by zhangstar333** [comment]
run buildall

**Review by HappenLee — DISMISSED** [review]
LGTM

**Comment by zhangstar333** [comment]
run buildall

**Comment by zhangstar333** [comment]
run buildall

**Review by yiguolei — APPROVED** [review]
LGTM

**Review by Gabriel39 — APPROVED** [review]
LGTM

**Pull request merged**